draft-ietf-httpbis-resumable-upload.md

```diff
@@ -406,6 +410,8 @@ The upload resource URL is the identifier used for modifying the upload. Without
 
 Some servers or intermediaries provide scanning of content uploaded by clients. Any scanning mechanism that relies on receiving a complete file in a single request message can be defeated by resumable uploads because content can be split across multiple messages. Servers or intermediaries wishing to perform content scanning SHOULD consider how resumable uploads can circumvent scanning and take appropriate measures. Possible strategies include waiting for the upload to complete before scanning a full file, or disabling resumable uploads.
 
+A malicious client may create upload resources and keep them alive by regularly sending `PATCH` requests with no or small content to the upload resources. The server SHOULD protect against such resource exhaustion attacks by limiting the lifetime of upload resources or the number of incomplete upload resources per client.
```

Comment:
This is good. Its reminds me a bit of the slow loris treatment in https://datatracker.ietf.org/doc/html/rfc9000#section-21.6, so let's ~steal~ borrow some of that

We should cite [SLOWLORIS ](https://web.archive.org/web/20150315054838/http://ha.ckers.org/slowloris/) but I'm too lazy right now and in a web editor.

```suggestion
Resumable uploads are vulnerable to Slowloris-style attacks [SLOWLORIS]. A malicious client may create upload resources and keep them alive by regularly sending `PATCH` requests with no or small content to the upload resources. This could be abused to exhaust server resources by creating and holding open uploads indefinately with minimal work.

Servers SHOULD provide mitigations for Slowloris attacks, such as increasing the maximum number of clients the server will allow, limiting the number of uploads a single client is allowed to make, imposing restrictions on the minimum transfer speed an uploads is allowed to have, and restricting the length of time an upload resource can exist.
```

Reply:
Is it common to refer to specific attacks in a spec like this? It's probably not the case with Slowloris but I imagine it being possible that a referenced attack is no longer applicable due to other reasons?

Reply:
It's probably better to name and link to similar attacks to help the reader understand the topic and related issues better. So it doesn't do any harm, I guess.